### PR TITLE
Exchange data historical api

### PR DESCRIPTION
**Pull request by tspenov**
```graphql
  subscription($exchange: String!, $tickerPair: String) {
    exchangeTrades(exchange: $exchange, tickerPair: $tickerPair) {
      exchange
      tickerPair
      datetime
      amount
      price
      cost
    }
  }
```

```graphql
  subscription($exchange: String!, $tickerPair: String) {
    exchangeMarketDepth(exchange: $exchange, tickerPair: $tickerPair) {
      exchange
      tickerPair
      datetime
      ask
      bid
      asks025PercentDepth
      asks025PercentVolume
      bids025PercentDepth
      bids025PercentVolume
    }
  }
```
```graphql
    {
      lastExchangeTrades(exchange: "Kraken", tickerPair: "ETH/EUR", limit: 100) {
        exchange
        tickerPair
        datetime
        amount
        price
        cost
      }
    }
```

```graphql
    {
      exchangeTrades(exchange: "Kraken", tickerPair: "ETH/EUR", from: "2019-10-20T00:00:00Z",  to: "2019-10-21T00:00:00Z") {
        exchange
        tickerPair
        datetime
        amount
        price
        cost
      }
    }
```

```graphql
    {
      exchangeTrades(exchange: "Kraken", tickerPair: "ETH/EUR", from: "2019-10-20T00:00:00Z",  to: "2019-10-21T00:00:00Z", interval: "1d") {
        exchange
        tickerPair
        datetime
        amount
        price
        cost
      }
    }
```

### CODE REVIEW

File: lib/sanbase_web/graphql/schema/queries/exchange_queries.ex [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+  alias SanbaseWeb.Graphql.Resolvers.ExchangeResolver
+
+  object :exchange_queries do
+    field :last_exchange_market_depth, list_of(:exchange_market_depth) do
+      arg(:exchange, non_null(:string))
+      arg(:ticker_pair, non_null(:string))
+      arg(:limit, non_null(:integer), default_value: 100)
+
+      resolve(&ExchangeResolver.last_exchange_market_depth/3)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/queries/exchange_queries.ex]: cache :?

File: lib/sanbase_web/graphql/schema/subscriptions/kafka_subscriptions.ex [hunk 2]
```diff
@@ -1,32 +1,30 @@
...
   use Absinthe.Schema.Notation
 
-  import_types(SanbaseWeb.Graphql.Schema.KafkaTypes)
-
   object :kafka_subscriptions do
     field :exchange_market_depth, :exchange_market_depth do
-      arg(:source, non_null(:string))
-      arg(:symbol, :string)
+      arg(:exchange, non_null(:string))
+      arg(:ticker_pair, :string)
```
Comment (IvanIvanoff) [inline lib/sanbase_web/graphql/schema/subscriptions/kafka_subscriptions.ex]: I think not, but is this API used anywhere? Because this is a breaking change

Comment (tspenov) [reply lib/sanbase_web/graphql/schema/subscriptions/kafka_subscriptions.ex]: Yes, not used.

File: lib/sanbase_web/graphql/schema/queries/exchange_queries.ex (hunk 1 above)
Comment (tspenov) [reply lib/sanbase_web/graphql/schema/queries/exchange_queries.ex]: added `cache_resolve`.

File: lib/sanbase/clickhouse/exchanges/market_depth.ex [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+      bids_20_percent_volume,
+      bids_2_percent_depth,
+      bids_2_percent_volume,
+      bids_30_percent_depth,
+      bids_30_percent_volume,
+      bids_5_percent_depth,
+      bids_5_percent_volume
+    FROM #{@table}
+    PREWHERE
+      source == ?1 AND symbol == ?2
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/exchanges/market_depth.ex]: we're using a single `=` throughout the codebase (they're equivalent)

File: lib/sanbase/clickhouse/exchanges/trades.ex [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+          cost: total_cost,
+          price: avg_price
+        }
+    end)
+  end
+
+  defp last_exchange_trades_query(exchange, ticker_pair, limit) do
+    query = """
+    SELECT
+      source, symbol, toUnixTimestamp(dt), side, amount, price, cost
```
Comment (IvanIvanoff) [inline lib/sanbase/clickhouse/exchanges/trades.ex]: we're consistently placing the timestamp as the first param in almost all queries

**Pull request merged**